js/newTab.js

```diff
@@ -119,6 +119,10 @@
   // Document is loaded. Initialize the New Tab Page.
   gNewTab.init();
   document.addEventListener("NewTabCommandReady", async(function* () {
+    let sw = (yield navigator.serviceWorker.ready).active;
```

Comment:
Actually the 'activate' event in service workers doesn't seem ideal for this. According to this doc: https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API/Using_Service_Workers#Basic_architecture (#6)
It looks like activate is only triggered when an updated version of the service worker exists. So since directoryLinksProvider handles its own timestamp etc. (at the moment) it needs to check the timestamp more often than just on activate. However, it sounds like your response cache headers patch will resolve this issue.